arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,59 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 300; //set the distance to 300
+const int fSpeed   = 70;  // 70% of the full speed forward
+const int bSpeed   = -70; // 70% of the full speed backward
+const int lDegrees = -75; // degrees to turn left
+const int rDegrees = 75;  // degrees to turn right
+bool flag = false;
+
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+SimpleCar car(control);
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+void autoStop(){
+  char input = Serial.read();
+  if(front.getDistance()>0 and front.getDistance()<150 and !flag){
```

Comment:
That's impossible due to the limitations of the car in the module, it would need way more speed.
And in the if statement it's a dual condition (the reading needs to be 150>reading<0) otherwise it won't brake, or does this code work differently from Java? We ran it in the module and we never saw the problem you're telling us about.

Reply:
Ahh we realized the problem but it's fun that we never saw this in the module when running it

Reply:
:+1:

Reply:
Fixed it(you can check again)